substrate/client/network/sync/src/lib.rs

```diff
@@ -206,24 +206,65 @@ struct GapSync<B: BlockT> {
 
 /// Action that [`engine::SyncingEngine`] should perform after reporting imported blocks with
 /// [`ChainSync::on_blocks_processed`].
-enum BlockRequestAction<B: BlockT> {
+pub enum BlockRequestAction<B: BlockT> {
```

Comment:
Why do we need to make this public?

Reply:
It's consumed by `SyncingEngine`, that owns `ChainSync`.

Reply:
But isn't this in the same crate?

Reply:
Technically, `SyncingEngine` is in the same root module as `ChainSync`, because `ChainSync` is defined in `lib.rs`. So `pub` is not needed at all (even as `pub(crate)`). Semantically, these types are public for `SyncingEngine`, so I feel inclined to highlight this with either `pub` or `pub(crate)`.

`pub(crate)` is probably a better option, but may be it's also worth moving `ChainSync` to a separate `chain_sync.rs` file.

@bkchr would it be better to move `ChainSync` and its types to a separate file and use `pub(crate)` for types consumed by `SyncingEngine`?

Reply:
> @bkchr would it be better to move `ChainSync` and its types to a separate file and use pub(crate) for types consumed by SyncingEngine?

Done. PR looks scary now due to moved code, but overall situation with `ChainSync` should have improved.

Reply:
@bkchr @altonen Could you have a look if moving `ChainSync` to a separate file worth it?

Reply:
> `pub(crate)` is probably a better option, but may be it's also worth moving `ChainSync` to a separate `chain_sync.rs` file.

Yeah sorry for these dumb questions :P I should have realized that the module is private to the crate. Pub is fine by me here.

Reply:
Done

Reply:
I was thinking this reordering could've been done in another PR but whatever. `SyncingEngine` should be moved to `sync/src/lib.rs` but that can be done in another PR.

Reply:
Why `SyncingEngine` should be moved to the root module and not kept in it's own module, may be reexporting it in the root module?